goon/browserassets/js/browserOutput.js

```diff
@@ -0,0 +1,944 @@
+			atBottom = true;
+			if ($('#newMessages').length) {
+				$('#newMessages').remove();
+			}
+		//If not, put the new messages box in
+		} else {
+			if ($('#newMessages').length) {
+				var messages = $('#newMessages .number').text();
+				messages = parseInt(messages);
+				messages++;
+				$('#newMessages .number').text(messages);
+				if (messages == 2) {
+					$('#newMessages .messageWord').append('s');
+				}
+			} else {
+				$messages.after('<a href="#" id="newMessages"><span class="number">1</span> new <span class="messageWord">message</span> <i class="icon-double-angle-down"></i></a>');
+			}
+		}
+	}
+
+	//Url stuff
+	if (message.length && flag != 'preventLink') {
+		message = linkify(message);
+	}
+
+	opts.messageCount++;
+
+	//Pop the top message off if history limit reached
+	if (opts.messageCount >= opts.messageLimit) {
+		$messages.children('div.entry:first-child').remove();
+		opts.messageCount--; //I guess the count should only ever equal the limit
+	}
+
+	//Actually append the message
+	var entry = document.createElement('div');
+	entry.className = 'entry';
+
+	if (filteredOut) {
+		entry.className += ' hidden';
+		entry.setAttribute('data-filter', filteredOut);
+	}
+
+	entry.innerHTML = message;
+	$messages[0].appendChild(entry);
+
+	//Actually do the snap
+	if (!filteredOut && atBottom) {
+		$('body,html').scrollTop($messages.outerHeight());
+	}
+
+	//Stuff we can do after the message shows can go here, in the interests of responsiveness
+	if (opts.highlightTerms && opts.highlightTerms.length > 0) {
+		highlightTerms(entry);
+	}
+}
+
+function internalOutput(message, flag)
+{
+	output(escaper(message), flag)
+}
+
+//Runs a route within byond, client or server side. Consider this "ehjax" for byond.
+function runByond(uri) {
+	window.location = uri;
+}
+
+function setCookie(cname, cvalue, exdays) {
+	cvalue = escaper(cvalue);
+	var d = new Date();
+	d.setTime(d.getTime() + (exdays*24*60*60*1000));
+	var expires = 'expires='+d.toUTCString();
+	document.cookie = cname + '=' + cvalue + '; ' + expires;
+}
+
+function getCookie(cname) {
+	var name = cname + '=';
+	var ca = document.cookie.split(';');
+	for(var i=0; i < ca.length; i++) {
+	var c = ca[i];
+	while (c.charAt(0)==' ') c = c.substring(1);
+		if (c.indexOf(name) === 0) {
+			return decoder(c.substring(name.length,c.length));
+		}
+	}
+	return '';
+}
+
+function rgbToHex(R,G,B) {return toHex(R)+toHex(G)+toHex(B);}
+function toHex(n) {
+	n = parseInt(n,10);
+	if (isNaN(n)) return "00";
+	n = Math.max(0,Math.min(n,255));
+	return "0123456789ABCDEF".charAt((n-n%16)/16) + "0123456789ABCDEF".charAt(n%16);
+}
+
+function changeMode(mode) {
+	switch (mode) {
+		case 'geocities':
+			//switch in stylesheet
+			opts.chatMode = mode;
+			break;
+		case 'console':
+
+			opts.chatMode = mode;
+			break;
+		case 'default':
+		default:
+			//remove loaded stylesheet/s
+			opts.chatMode = 'default';
+	}
+}
+
+function handleClientData(ckey, ip, compid) {
```

Comment:
If you guys end up porting this specific function as well (vg did not) make sure you at least change the following

- `?_src_=chat&proc=analyzeClientData` will show up in the href logs, make sure you actually remove the cid/ip from the public logs, as /vg/ did not which is bad